src/classes/DoubleLinkedList.js

```diff
@@ -0,0 +1,77 @@
+class DoubleLinkedList {
+    constructor(){
+        this.first = null;
+        this.last = null;
+        this.len = 0;
+    }
+
+    push(node) {
+        this.len += 1;
+        if (this.first === null) {
+            this.first = node;
+            this.last = node;
+        } else {
+            this.last.next = node;
+            node.prev = this.last;
+            this.last = node;
+        }
+    }
+
+    unshift(node) {
+        this.len += 1;
+        if (this.first === null) {
+            this.first = node;
+            this.last = node;
+        } else {
+            this.first.prev = node;
+            node.next = this.first;
+            this.first = node;
+        }
+    }
+
+    pop() {
+        this.len -= 1;
+        let node = null;
+        if (this.first !== null) {
+            if (this.first == this.last) {
+                node = this.first;
+                this.first = null;
+                this.last = null;
+                return node;
+            } else {
+                node = this.last;
+                this.last = this.last.prev;
+                this.last.next = null;
+            }
+        }
+        return node;
+    }
```

Comment:
the way you put conditions into conditions, you increase the difficult to read the code. What do you think about this?

```suggestion
    pop() {
        if (!this.first) return;

        const node = this.last;
        this.last = this.last.prev;
        if(!this.last) this.first = null;
        this.len -= 1;

        return node;
    }
```